File: go-sdk/pkg/models/shared/updateclientrequest.go
package shared

type UpdateClientRequest struct {
	Description            *string                `json:"description,omitempty"`
	Metadata               map[string]interface{} `json:"metadata,omitempty"`
	Name                   string                 `json:"name"`
	PostLogoutRedirectUris []string               `json:"postLogoutRedirectUris,omitempty"`
	Public                 *bool                  `json:"public,omitempty"`
	RedirectUris           []string               `json:"redirectUris,omitempty"`
	Trusted                *bool                  `json:"trusted,omitempty"`
}

File: ts-sdk/src/sdk/models/shared/updateclientrequest.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose } from "class-transformer";


export class UpdateClientRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "description" })
  description?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata()
  @Expose({ name: "name" })
  name: string;

  @SpeakeasyMetadata()
  @Expose({ name: "postLogoutRedirectUris" })
  postLogoutRedirectUris?: string[];

  @SpeakeasyMetadata()
  @Expose({ name: "public" })
  public?: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "redirectUris" })
  redirectUris?: string[];

  @SpeakeasyMetadata()
  @Expose({ name: "trusted" })
  trusted?: boolean;
}